fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryIndexPlan.java

```diff
@@ -115,42 +113,38 @@ public RecordQueryIndexPlan(@Nonnull final String indexName,
-        this(indexName, commonPrimaryKey, scanParameters, useIndexPrefetch, reverse, strictlySorted, Optional.of(matchCandidate), resultType);
+        this(indexName, commonPrimaryKey, scanParameters, indexFetchMethod, reverse, strictlySorted, Optional.of(matchCandidate), resultType);
     }
 
     private RecordQueryIndexPlan(@Nonnull final String indexName,
                                  @Nullable final KeyExpression commonPrimaryKey,
                                  @Nonnull final IndexScanParameters scanParameters,
-                                 @Nonnull final RecordQueryPlannerConfiguration.IndexPrefetchUse useIndexPrefetch,
+                                 @Nonnull final RecordQueryPlannerConfiguration.IndexFetchMethod indexFetchMethod,
                                  final boolean reverse,
                                  final boolean strictlySorted,
                                  @Nonnull final Optional<? extends ScanWithFetchMatchCandidate> matchCandidateOptional,
                                  @Nonnull final Type resultType) {
         this.indexName = indexName;
         this.commonPrimaryKey = commonPrimaryKey;
         this.scanParameters = scanParameters;
-        this.useIndexPrefetch = useIndexPrefetch;
+        this.indexFetchMethod = indexFetchMethod;
         this.reverse = reverse;
         this.strictlySorted = strictlySorted;
         this.matchCandidateOptional = matchCandidateOptional;
         this.resultType = resultType;
-        if (useIndexPrefetch != RecordQueryPlannerConfiguration.IndexPrefetchUse.NONE) {
+        if (indexFetchMethod != RecordQueryPlannerConfiguration.IndexFetchMethod.SCAN_AND_FETCH) {
             if (commonPrimaryKey == null) {
-                error("Index Prefetch cannot be used without a primary key. Falling back to regular scan.");
-                this.useIndexPrefetch = RecordQueryPlannerConfiguration.IndexPrefetchUse.NONE;
+                logWarning("Index remote fetch cannot be used without a primary key. Falling back to regular scan.");
+                this.indexFetchMethod = RecordQueryPlannerConfiguration.IndexFetchMethod.SCAN_AND_FETCH;
             }
             if (scanParameters.getScanType() != IndexScanType.BY_VALUE) {
-                error("Index Prefetch can only be used with VALUE index scan. Falling back to regular scan.");
-                this.useIndexPrefetch = RecordQueryPlannerConfiguration.IndexPrefetchUse.NONE;
-            }
-            if (!FDBDatabaseFactory.instance().getDatabase().getAPIVersion().isAtLeast(APIVersion.API_VERSION_7_1)) {
-                error("Index Prefetch can only be used with API_VERSION of at least 7.1. Falling back to regular scan.");
-                this.useIndexPrefetch = RecordQueryPlannerConfiguration.IndexPrefetchUse.NONE;
+                logWarning("Index remote fetch can only be used with VALUE index scan. Falling back to regular scan.");
```

Comment:
Is there planner logic to not generate plans with an `IndexFetchMethod` of `REMOTE_FETCH` if the index scan type isn't `BY_VALUE`? Because those would issue a lot of `warn` logs if there's not such a protection.

Reply:
That is a good point. There is no such logic. I do wonder though whether it is a good idea to "pollute" the planner code with specific business logic that looks at the kind of index and changes the plan configuration. It may be that in this case the plan should just ignore that parameter "silently", or maybe with a `debug/info` message since this can be considered a case where the system configuration does not apply to what the plan can do so the plan does what it can, no harm done...

Reply:
Yeah, I feel like a silent switch to `SCAN_AND_FETCH` is probably better than a `WARN` message here, given how common I think that occurrence might be. (There's probably a similar case to be made about the `commonPrimaryKey == null` check.)

That being said, I don't think it's necessarily wrong for the planner to take into account the capabilities of the index and adjust the planner configuration accordingly, but my opinions on what the planner should/shouldn't do aren't informed by a lot of familiarity with planner internals

Reply:
Changed both logs to `DEBUG`. Did not change the planner (I believe this will tighten the dependency between the two).